File: apps/bench/app/controller.rb
require 'dm-serializer'
require 'sinatra'
require 'haml'
require 'user_agent'

class Controller < Sinatra::Base

  set :static, true
  set :public_folder, PUBLIC_DIR

  get '/' do
    haml :index
  end

  get '/monitor' do
    haml :monitor
  end

  get '/results' do
    results = Result.all
    results &= Result.all(:type => params[:type]) if (params[:type])
    results &= Result.all(:user_agent.like => '%' + params[:user_agent] + '%') if (params[:user_agent])
    results &= Result.all(:dht_size => params[dht_size].to_i) if (params[:dht_size])
    results.to_json
  end

  post '/results' do
    uagt = UserAgent.new(request.user_agent)
    infos = {
      :user_agent => [uagt.name, uagt.engine, uagt.platform].join(','),
      :mobile     => uagt.mobile?,
      :dht_size   => ENV['DHT_SIZE'].to_i,
      :created_at => Time.now
    }

    results = JSON.parse(request.body.read)
    results.each { |r| Result.create(r.merge(infos)) }
    "OK."
  end
end

File: apps/bench/app/controller.js
var express = require('express'),
    uagent  = require('express-useragent'),
    _       = require('underscore');

var Result = app.settings.db.import(__dirname+'/models/result.js');

app.use(express.bodyParser())
   .use(uagent.express());

app.get('/', function(req, res) {
  res.render('index');
});

app.get('/monitor', function(req, res) {
  res.render('monitor');
});

app.post('/results', function(req, res) {
  var infos = {
    user_agent : [req.useragent.Browser, req.useragent.OS].join(','),
    mobile     : req.useragent.isMobile,
    dht_size   : app.settings.dht.size,
    created_at : Date.now()
  };

  _.each(req.body, function(r) {
    Result.create(_.extend(r, infos));
  });
  res.send();
});

app.get('/results', function(req, res) {
  Result.all()
  .success(function(results) {
    res.json(results);
  }).error(function(err) {
    res.send(err, 500);
  })
});